File: Number System Conversion/Binary - Decimal/Bin2Dec.py
def convert(n):
    dec,i=0,0
    while n!=0:
        rem=n%10
        n//=10
        dec+=rem* pow(2,i)
        i+=1
    return dec

n=int(input("Enter a binary number : "))
print("{} in binary = {} in decimal".format(n,convert(n)))

File: Number System Conversion/Binary - Decimal/Bin2Dec.java
import java.util.*;
class Bin2Dec
{
    int convert(int n) {
        int dec = 0, i = 0, rem;
        while (n != 0) {
            rem = n % 10;
            n /= 10;
            dec += rem * (int)Math.pow(2, i);
            ++i;
        }
        return dec;
    }
    public static void main(String[] args)
    {
        Scanner sc=new Scanner (System.in);
        System.out.print("Enter a binary number ");
        int n=sc.nextInt();
        sc.close();
        Bin2Dec ob=new Bin2Dec();
        System.out.print(n+" in binary = "+ob.convert(n)+" in decimal");   
    }
}